NestJS/src/Event/event.listener.ts

```diff
@@ -4,14 +4,75 @@ import { Controller } from '@nestjs/common';
 @Controller('event')
 export class EventController {
   constructor(private readonly eventService: EventService) {
-
-     var exchange =  connection.declareExchange("hello", 'direct');
-     var queue = connection.declareQueue("event", {durable: true});
-     queue.bind(exchange, 'black');
-     queue.activateConsumer((message) => {
-         console.log("Message received: " + JSON.parse(message.getContent()));
-     }, {noAck: true});
+    var exchange =  connection.declareExchange("ExchangeName", 'direct');
+    var queue = connection.declareQueue("event", {durable: true});
+    queue.bind(exchange, 'black');
+    queue.activateConsumer(async (message) => {
+         let json = JSON.parse(message.getContent());
+         await run(json, this.eventService, function(err, response){
+           let msg = {
+             "title": json.title,
+             "body": {}
+           }
+           msg.body = response;
+           console.log("Message received: ");
+           console.log(msg);
+      });
+    }, {noAck: true});
+  }
+}
+async function run(json, eventService, callback){
+  if(json.title == "create"){
+    await eventService.createEvent(json, function(err, response){
```

Comment:
this callback is repeated, in this case you can use separate function

Reply:
ok,I will change it